pandas/tests/extension/decimal/test_decimal.py

```diff
@@ -478,3 +478,24 @@ def DecimalArray__my_sum(self):
     s = pd.Series(DecimalArray(data))
     result = s.groupby(np.array([0, 0, 0, 1, 1])).agg(lambda x: x.values.my_sum())
     tm.assert_series_equal(result, expected, check_names=False)
+
+
+def test_indexing_no_materialize(monkeypatch):
+    # See https://github.com/pandas-dev/pandas/issues/29708
+    # Ensure that indexing operations do not materialize (convert to a numpy
+    # array) the ExtensionArray unnecessary
+
+    def DecimalArray__array__(self, dtype=None):
+        raise Exception("tried to convert a DecimalArray to a numpy array")
```

Comment:
This is not fully fool proof, as such an Exception could be catched somewhere in the indexing code. But I at least verified that for the `s[s > 0.5]` case, this test actually failed as desired before making the fix.